### PR TITLE
Add ssl support for the probe

### LINKED ISSUE #559: Probe posting should follow redirects

### PR DESCRIPTION
**Pull request by paulbellamy**
Hosts ending in `:443` are treated as SSL.

There is also a `--probe.insecure` flag, for using it in development
environments where SSL cannot be verified.

Probably needs a unit test for the HTTPPublisher written. Will do that if the code/concept looks ok.

Fixes #559 in a way.

### CODE REVIEW

File: xfer/http_publisher.go [hunk 1]
```diff
@@ -15,24 +17,39 @@ type HTTPPublisher struct {
...
-func NewHTTPPublisher(target, token, probeID string) (string, *HTTPPublisher, error) {
+func NewHTTPPublisher(target, token, probeID string, insecure bool) (string, *HTTPPublisher, error) {
+	_, port, err := net.SplitHostPort(target)
+	if err != nil {
+		return "", nil, err
+	}
+	scheme := "http"
+	if port == "443" {
+		scheme = "https"
+	}
```
Comment (tomwilkie) [inline xfer/http_publisher.go]: Should we push this logic into sanitise.URL?

**Comment by tomwilkie** [comment]
Do we have the certificate bundle installed in the scope container for this to work?

**Comment by paulbellamy** [comment]
Oh, for an authority to validate certs. Probably not.

File: xfer/http_publisher.go (hunk 1 above)
Comment (paulbellamy) [inline xfer/http_publisher.go]: yes, probably, actually.

**Comment by tomwilkie** [comment]
LGTM

**Pull request merged**